os/arch/arm/src/s5j/sss/security_hal_wrapper.c

```diff
@@ -0,0 +1,1632 @@
+		}
+		*--(*p) = (unsigned char)len;
+		return 1;
+	}
+
+	if (len <= 0xFF) {
+		if (*p - start < 2) {
+			return -HAL_INVALID_ARGS;
+		}
+		*--(*p) = (unsigned char)len;
+		*--(*p) = 0x81;
+		return 2 ;
+	}
+
+	if (len <= 0xFFFF) {
+		if (*p - start < 3) {
+			return -HAL_INVALID_ARGS;
+		}
+		*--(*p) = (len) & 0xFF;
+		*--(*p) = (len >>  8) & 0xFF;
+		*--(*p) = 0x82;
+		return 3;
+	}
+
+	if (len <= 0xFFFFFF) {
+		if (*p - start < 4) {
+			return -HAL_INVALID_ARGS;
+		}
+
+		*--(*p) = (len) & 0xFF;
+		*--(*p) = (len >> 8) & 0xFF;
+		*--(*p) = (len >> 16) & 0xFF;
+		*--(*p) = 0x83;
+		return 4;
+	}
+
+	if (len <= 0xFFFFFFFF) {
+		if (*p - start < 5) {
+			return -HAL_INVALID_ARGS;
+		}
+
+		*--(*p) = (len) & 0xFF;
+		*--(*p) = (len >> 8) & 0xFF;
+		*--(*p) = (len >> 16) & 0xFF;
+		*--(*p) = (len >> 24) & 0xFF;
+		*--(*p) = 0x84;
+		return 5;
+	}
+
+	return -HAL_INVALID_ARGS;
+}
+
+static int hal_asn1_write_mpi(unsigned char **p, unsigned char *start, const hal_mpi *X)
+{
+	int ret;
+	uint32_t len = 0;
+
+	len = hal_mpi_size(X);
+
+	if (*p < start || (uint32_t)(*p - start) < len) {
+		return -HAL_INVALID_ARGS;
+	}
+
+	(*p) -= len;
+	HAL_MPI_CHK(hal_mpi_write_binary(X, *p, len));
+
+	if (X->s == 1 && **p & 0x80) {
+		if (*p - start < 1) {
+			return HAL_NOT_ENOUGH_MEMORY;
+		}
+		*--(*p) = 0x00;
+		len += 1;
+	}
+
+	HAL_ASN1_CHK_ADD(len, hal_asn1_write_len(p, start, len));
+	HAL_ASN1_CHK_ADD(len, hal_asn1_write_tag(p, start, 0x02));
+
+	ret = (int)len;
+
+cleanup:
+	return ret;
+}
+
+static int hal_ecdsa_signature_to_asn1(const hal_mpi *r, const hal_mpi *s, unsigned char *sig, uint32_t *slen)
+{
+	int ret;
+	unsigned char buf[HAL_MAX_ECDSA_LEN];
+	unsigned char *p = buf + sizeof(buf);
+	uint32_t len = 0;
+
+	HAL_ASN1_CHK_ADD(len, hal_asn1_write_mpi(&p, buf, s));
+	HAL_ASN1_CHK_ADD(len, hal_asn1_write_mpi(&p, buf, r));
+	HAL_ASN1_CHK_ADD(len, hal_asn1_write_len(&p, buf, len));
+	HAL_ASN1_CHK_ADD(len, hal_asn1_write_tag(&p, buf, 0x20|0x10));
+
+	memcpy(sig, p, len);
+	*slen = len;
+
+	return HAL_SUCCESS;
+}
+
+static int hal_asn1_get_mpi(unsigned char **p, const unsigned char *end, hal_mpi *X)
+{
+	int ret;
+	uint32_t len;
+
+	if ((ret = hal_asn1_get_tag(p, end, &len, 0x02)) != 0) {
+		return ret;
+	}
+
+	ret = hal_mpi_read_binary(X, *p, len);
+
+	*p += len;
+
+	return ret;
+}
+
+/**
+ * Common
+ */
+int hal_init(hal_init_param *params)
+{
+	HWRAP_ENTER;
+	return HAL_NOT_SUPPORTED;
+}
+
+int hal_deinit(void)
+{
+	HWRAP_ENTER;
+	return HAL_NOT_SUPPORTED;
+}
+
+int hal_free_data(hal_data *data)
+{
+	HWRAP_ENTER;
+	if (data) {
+		if (data->data) {
+			free(data->data);
+		}
+		if (data->priv) {
+			free(data->priv);
+		}
+	}
+	return HAL_SUCCESS;
+
+}
+
+int hal_get_status(void)
+{
+	HWRAP_ENTER;
+	return isp_get_status();
+}
+
+/**
+ * Key Manager
+ */
+int hal_set_key(hal_key_type mode, uint32_t key_idx, hal_data *key, hal_data *prikey)
+{
+	HWRAP_ENTER;
+	if (prikey != NULL) {
+		return HAL_NOT_SUPPORTED;
+	}
+
+	unsigned int key_type;
```

Comment:
How about rename 'hal_get_key_type' to hal_get_ss_type_from_key_type
and then use it here. and second param is not neccessary, you can change return type.